packages/react-ui-kit/src/Form/SelectContainer.tsx

```diff
@@ -0,0 +1,290 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+/** @jsx jsx */
+import {CSSObject, jsx} from '@emotion/react';
+import React, {FormEvent, MouseEvent, ReactElement, useRef, useState} from 'react';
+
+import {selectStyle, dropdownStyles, dropdownOptionStyles} from './Select.styles';
+import {Checkbox} from './Checkbox';
+import InputLabel from './InputLabel';
+import type {Theme} from '../Layout';
+import {filterProps} from '../util';
+import {ArrowDown} from '../Icon/ArrowDown';
+import useOutsideClick from '../Misc/useOutsideClick';
+
+export type SelectOption = {
+  value: string | number;
+  label: string;
+  description?: string;
+  isDisabled?: boolean;
+};
+
+export interface CommonSelectProps<T extends SelectOption = SelectOption, K extends SelectOption = SelectOption> {
+  id: string;
+  dataUieName: string;
+  options: T[];
+  onOptionChange: (idx: number) => void;
+  helperText?: string;
+  label?: string;
+  disabled?: boolean;
+  required?: boolean;
+  markInvalid?: boolean;
+  error?: ReactElement;
+  wrapperCSS?: CSSObject;
+  isMultiSelect?: boolean;
+  selectedLabel?: string;
+  selectedValue?: string;
+  checkIsSelected?: (optionValue: string | number) => boolean;
+}
+
+const filterSelectProps = props => filterProps(props, ['markInvalid']);
+
+const placeholderText = '- Please select -';
+
+export const SelectContainer = <T extends SelectOption = SelectOption>({
+  id,
+  label,
+  error,
+  helperText,
+  options = [],
+  onOptionChange,
+  required,
+  markInvalid,
+  dataUieName,
+  wrapperCSS = {},
+  isMultiSelect = false,
```

Comment:
not really a full idea but maybe this can give you some idea of what i had in mind. 

options: SelectOption[] | multiSelectOption[]

const singleselectoptions = options.map((x)=>{
<SelectOption options={option} onClick={selectOptionclick}/>}

const multiselectoptions = options.map((x)=>{
<MultiSelectOption options={options} onclick={selectMultiOptionClick}/>}

//SingleSelect
SelectContainer options={singleselectoptions} ...
...
//MultiSelect
SelectContainer options=[multiselectoptions} ...